OpenRA.Mods.Common/ServerTraits/LobbyCommands.cs

```diff
@@ -269,37 +269,41 @@ public bool InterpretCommand(S server, Connection conn, Session.Client client, s
 							return true;
 						}
 
-						if (!ValidateSlotCommand(server, conn, client, parts[0], true))
+						var slotKey = parts[0];
+
+						if (!ValidateSlotCommand(server, conn, client, slotKey, true))
 							return false;
 
-						var slot = server.LobbyInfo.Slots[parts[0]];
-						var bot = server.LobbyInfo.ClientInSlot(parts[0]);
+						var slot = server.LobbyInfo.Slots[slotKey];
+						var bot = server.LobbyInfo.ClientInSlot(slotKey);
 						int controllerClientIndex;
 						if (!Exts.TryParseIntegerInvariant(parts[1], out controllerClientIndex))
 						{
 							Log.Write("server", "Invalid bot controller client index: {0}", parts[1]);
 							return false;
 						}
 
-						var botType = parts.Skip(2).JoinWith(" ");
-
 						// Invalid slot
 						if (bot != null && bot.Bot == null)
 						{
 							server.SendOrderTo(conn, "Message", "Can't add bots to a slot with another client.");
 							return true;
 						}
 
+						var botType = parts.Skip(2).JoinWith(" ");
+						var slotIdx = server.LobbyInfo.Slots.Keys.ToArray().IndexOf(slotKey);
+						var botName = "{0} {1}".F(botType, slotIdx + 1);
```

Comment:
Doing this in the lobby isn't great, IMO, and labelling them with a number based on the slot is *really* not great.  If you have a match with 4 real people and then one bot it will be called "Rush AI 5".

A better solution would be to adjust the names in the Player constructor, numbering them based on the number of other bots of the same type in the match.  E.g if you have a game with a single Rush AI then it will be called "Rush AI", but if you have a game with two, then they should be called "Rush AI 1" and "Rush AI 2" regardless of the specific slots they are in.